File: src/main/java/com/example/springpractice/product/dto/ProductResponse.java
package com.example.springpractice.product.dto;

import com.example.springpractice.product.Product;

public record ProductResponse(
        Long id,
        String name,
        int price,
        int stockQuantity
) {
    public static ProductResponse from(Product product) {
        return new ProductResponse(
                product.getId(),
                product.getName(),
                product.getPrice(),
                product.getStockQuantity()
        );
    }
}

File: kotlin-practice/src/main/kotlin/com/example/kotlinpractice/product/dto/ProductResponse.kt
package com.example.kotlinpractice.product.dto

import com.example.kotlinpractice.product.Product

data class ProductResponse(
    val id: Long?,
    val name: String,
    val price: Int,
    val stockQuantity: Int
) {
    companion object {
        fun from(product: Product): ProductResponse =
            ProductResponse(product.id, product.name, product.price, product.stockQuantity)
    }
}
